src/irmin/conf.ml

```diff
@@ -95,10 +95,13 @@ let key ?docs ?docv ?doc ~spec name ty default =
         | _ -> raise @@ Invalid_argument name)
       name
   in
-  let to_univ, of_univ = Univ.create () in
-  let k = { name; ty; default; to_univ; of_univ; doc; docv; docs } in
-  Spec.update spec name (K k);
-  k
+  if Spec.find_key spec name |> Option.is_some then
```

Comment:
Just following the style in this file of using `if-then-else` instead of `match`.

Reply:
I see the point for stylistic consistency. But maybe we should challenge these things once in a while.

I'd suggest using a `match` for the new code (but not updating the existing usages of `Option.is_some` in the file).

For historical context: This is the PR that introduced the usage of the `if-then-else`s: https://github.com/mirage/irmin/pull/1492 . There was no discussion on the usage of `if-then-else` in the PR. It was a larger PR and reviews focused on the bigger ideas.

Reply:
Updated to used `match`. Also, tidied the other code while I'm in this file (https://github.com/mirage/irmin/pull/2252/commits/ac1c72a38d0664f174f9412e7dbfb4576d059e48) since it's not that big of a change. Thanks for the nudge. :smile: